src/main/java/erika/Erika.java

```diff
@@ -0,0 +1,293 @@
+        if (description.trim().isEmpty()) {
+            throw new EmptyDescriptionException("Deadline");
+        }
+
+        Deadline newDeadline = new Deadline(description, byText);
+        tasks.add(newDeadline);
+        try {
+            fileSystem.appendTaskToFile(newDeadline);
+        } catch (IOException e) {
+            printMessage("Unable to append Deadline to FS. IO Error occurred");
+        }
+        return newDeadline;
+    }
+
+    private static void addTodo(String line) throws EmptyDescriptionException{
+        if (!line.contains("todo ")) {
+            throw new EmptyDescriptionException("Todo");
+        }
+        Todo newTodo = new Todo(line.substring(line.indexOf(" ")+Settings.SPACE_OFFSET));
+        tasks.add(newTodo);
+        try {
+            fileSystem.appendTaskToFile(newTodo);
+        } catch (IOException e) {
+            printMessage("Unable to append Todo to FS. IO Error occurred");
+        }
+        printAddedMessage(newTodo);
+    }
+
+    private static void indexOperation(String line, boolean isDelete) throws IndexOutOfBoundsException, EmptyDescriptionException{
+        if(!isDelete) {
+            markEntry(line);
+        } else {
+            deleteEntry(line);
+        }
+    }
+
+    private static void deleteEntry(String line) throws EmptyDescriptionException {
+        if (!line.contains("delete ")) {
+            throw new EmptyDescriptionException("delete");
+        }
+        markIndex = extractTaskIndex(line);
+        if (markIndex <= 0 || markIndex > Task.getTaskArraySize()) {
+            throw new IndexOutOfBoundsException();
+        }
+        printDeletedMessage();
+        tasks.remove(markIndex-1);
+        Task.decrementTaskArraySize();
+        saveChangesToFileSystem();
+    }
+
+    private static void markEntry(String line) throws EmptyDescriptionException {
+        if (!line.contains("mark ")) {
+            throw new EmptyDescriptionException("mark");
+        }
+        markIndex = extractTaskIndex(line);
+        if (markIndex <= 0 || markIndex > Task.getTaskArraySize()) {
+            throw new IndexOutOfBoundsException();
+        }
+        if (line.contains("unmark ")) {
+            tasks.get(markIndex - 1).setMark(false);
+            printUnmarkedMessage();
+        } else {
+            tasks.get(markIndex - 1).setMark(true);
+            printMarkedMessage();
+        }
+        saveChangesToFileSystem();
+    }
+
+    private static int extractTaskIndex(String line) throws NumberFormatException{
+        String digitString = line.replaceAll("[^0-9-]", "");
+        if (!digitString.isEmpty()) {
+            return Integer.parseInt(digitString);
+        } else {
+            throw new NumberFormatException();
+        }
+    }
+
+    private static void printMessage(String message) {
+        System.out.println("\t____________________________________________________________");

```

Comment:
Consider making this line into a constant string, for easy reproducibility and keeping visuals consistent